packages/plugin-kit/src/lib/DecodeError.ts

```diff
@@ -0,0 +1,22 @@
+import * as t from "io-ts";
```

Comment:
Maybe a general-purpose "support" package is inevitable, but I think it could be a dangerous move. By creating a "support" package, we make it easy for us to prematurely abstract code to be shared across packages. We may then run into technical problems, like cross-platform incompatibilities (e.g. importing Node.js packages into browser code) or making non-obvious breaking changes.

`@slicemachine/core` suffers from this problem today.

As much as possible, I think we should contain all code that a project uses within itself, except for very deliberate exports (e.g. `createSliceMachineManager()` from `@slicemachine/manager` is a deliberate export).

I can't think of another solution, but maybe we can refine the idea of a "support" package. Rather than making it generic in the same way that a "utils," "helpers," or "core" package would be, we can be deliberate about what the package is.

For example, maybe the package only contains codecs/validators. What is allowed in the package becomes more obvious and discourages bad practices. Such a package could contain `decode()`, `DecodeError`, and shared codecs.

Some ideas:
- `@slicemachine/validators`
- `@slicemachine/io-ts-validators`
- `@slicemachine/io-ts-codecs`
- `@slicemachine/io-ts-decoders`

What do you think?

Reply:
I guess codecs, would be limted to io-ts types, while validators could theoretically include form validators.... codecs might be better :)